Assets/Code/Scripts/NoclipRealityManagement/NoclipManager.cs

```diff
@@ -88,19 +89,27 @@ private IEnumerator DisableNoclip()
         yield return null;
     }
 
+
     /// <summary>
     /// Allows the player to enable/disable noclip mode.
     /// </summary>
     private void Update()
     {
-        if (Input.GetKeyDown(_noclipOptions.noclipKey) && _playerCanEnableNoclip)
-        {
-            StartCoroutine(EnableNoclip());
-        }
-
-        if (Input.GetKeyDown(_noclipOptions.noclipKey) && _playerCanDisableNoclip)
+        if (Input.GetKeyDown(_noclipOptions.noclipKey))
         {
-            StartCoroutine(DisableNoclip());
+            if (_playerCanEnableNoclip)
+            {
+                StartCoroutine(EnableNoclip());
+            } 
+            else
+            {
+                EventManager.TriggerEvent("DisplayHint", "NOCLIP ZONE NOT FOUND"); 
+            }
+            
+            if (_playerCanDisableNoclip)
+            {
+                StartCoroutine(DisableNoclip());
+            }
```

Comment:
Something like this should fix the bug and also give an additional and useful hint to the user. **Please try it, as I did not have time to test it.**
 
```suggestion
            if (_playerCanEnableNoclip)
                StartCoroutine(EnableNoclip());
            else if (_playerCanDisableNoclip)
                StartCoroutine(DisableNoclip());
            else if (!_noclipEnabled)
                EventManager.TriggerEvent("DisplayHint", "NOCLIP ZONE NOT FOUND. PRESSING P HAS NO EFFECT"); 
            else if (_noclipEnabled)
                EventManager.TriggerEvent("DisplayHint", "RETURN TO YOUR BODY TO DISABLE NOCLIP"); 
```